### PR TITLE
Fix EnhancedTracker not auto binding when adding manga to library

### LINKED ISSUE #306: Komga tracking not enabled by default when adding an entry to the library
### Steps to reproduce

1. Install de Komga extension (1.4.50) and setup
2. Explore the newly configured Komga server
3. Long tap on an entry to add it to the Mihon library
4. Open the newly added entry, tracking is not enabled by default

### Expected behavior

Step 3 should add the Komga's entry to Mihon library and enable tracking.

### Actual behavior

Step 3 adds the Komga's entry to Mihon library but does not enable tracking.

EDIT: I had the same issue with Tachiyomi.

### Crash logs

_No response_

### Mihon version

0.16.3

### Android version

Android 13

### Device

Pixel 4a

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have gone through the [FAQ](https://mihon.app/docs/faq/general) and [troubleshooting guide](https://mihon.app/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.16.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by brewkunz**
This PR is an attempt to fix the issue #306.

Tracking used to be enabled by default when adding an item to the library, which is not the case anymore.
This fix tries to bind with all logged In enhanced trackers upon adding an item to the library.

~~I think there still is an issue with the track being saved with a `last_read_chapter` value of 0.0, but the problem seems to lie in the `SyncChapterProgressWithTrack` and is not introduced by this fix.~~

### CODE REVIEW

File: app/src/main/java/eu/kanade/domain/track/interactor/AddTracks.kt [hunk 1]
```diff
@@ -79,19 +79,19 @@ class AddTracks(
 
     suspend fun bindEnhancedTrackers(manga: Manga, source: Source) = withNonCancellableContext {
         withIOContext {
-            getTracks.await(manga.id)
```
Comment (brewkunz) [inline app/src/main/java/eu/kanade/domain/track/interactor/AddTracks.kt:82]: This only returned already active tracks, which I don't think is intended.

File: app/src/main/java/eu/kanade/domain/track/interactor/AddTracks.kt [hunk 2]
```diff
@@ -79,19 +79,19 @@ class AddTracks(
...
+            trackerManager.loggedInTrackers()
                 .filterIsInstance<EnhancedTracker>()
                 .filter { it.accept(source) }
                 .forEach { service ->
                     try {
                         service.match(manga)?.let { track ->
                             track.manga_id = manga.id
                             (service as Tracker).bind(track)
-                            insertTrack.await(track.toDomainTrack()!!)
+                            insertTrack.await(track.toDomainTrack(idRequired = false)!!)
```
Comment (brewkunz) [inline app/src/main/java/eu/kanade/domain/track/interactor/AddTracks.kt:90]: This is now needed since the track might not have an id yet.

**Comment by AntsyLich** [comment]
Good job trying to fix your own issue

File: app/src/main/java/eu/kanade/domain/track/interactor/AddTracks.kt (hunk 1 above)
Comment (AntsyLich) [reply app/src/main/java/eu/kanade/domain/track/interactor/AddTracks.kt:82]: Seems like the previous maintainer fucked up during a cleanup

File: app/src/main/java/eu/kanade/domain/track/interactor/AddTracks.kt (hunk 2 above)
Comment (AntsyLich) [reply app/src/main/java/eu/kanade/domain/track/interactor/AddTracks.kt:90]: the old implementation didn't need this (which actually did auto bind) either so should be fine? unless it was a issue during testing

Comment (brewkunz) [reply app/src/main/java/eu/kanade/domain/track/interactor/AddTracks.kt:90]: It was indeed an issue during my tests. If the item you add to the library has no exisiting track with the tracker, the track has no id yet and `toDomainTrack` throws an exception.
It seems like this method allows for inserting and updating, but when you are inserting you have to set this parameter to false.

**Comment by AntsyLich** [comment]
> I think there still is an issue with the track being saved with a last_read_chapter value of 0.0, but the problem seems to lie in the SyncChapterProgressWithTrack and is not introduced by this fix.

Should be a separate PR

**Comment by brewkunz** [comment]
> > I think there still is an issue with the track being saved with a last_read_chapter value of 0.0, but the problem seems to lie in the SyncChapterProgressWithTrack and is not introduced by this fix.
> 
> Should be a separate PR

I moved this part to its new PR.

**Pull request merged**